### PR TITLE
Add eMMC RPMB key write lock functions

### PR DESCRIPTION
**Pull request by hbeberman**
Add two new weak functions for leaving CFG_RPMB_WRITE_KEY enabled in
production images. This resolves the need for a separate OP-TEE during
manufacturing as it will disable its own ability to write eMMC RPMB keys
after doing it once in the factory.

This additional write lock protection is necessary because a system in
the field that has had its eMMC replaced by an attacker will submit the
RPMB key in plaintext to the supplicant for provisioning the new eMMC.
An attacker could intercept this value and use it to decrypt the
original eMMC RPMB from the platform.

TEE_Result tee_otp_check_rpmb_key_write_lock(void)
Checks platform defined fuse to see if the SoC has already provisioned
an RPMB previously. If the return is not TEE_SUCCESS then the RPMB key
write will fail before it's transmitted in plaintext to the supplicant.

TEE_Result tee_otp_set_rpmb_key_write_lock(void)
Blows the platform defined fuse that's checked prior to writing the RPMB
key to the supplicant.

This is enabled on i.MX6 Quad platforms by enabling CFG_FSL_SEC=y
and CFG_RPMB_KEY_WRITE_LOCK=y. Before doing this you should ensure that
RPMB_KEY_WRITE_LOCK_FUSE_BITS and RPMB_KEY_WRITE_LOCK_FUSE_WORD are
correct for your platform. Some OEMs may already be using fuses from the
GP1 or GP2 fuse words.

RPMB write lock fusing is not currently supported on i.MX7

Signed-off-by: Henry Beberman <hebeberm@microsoft.com>

### CODE REVIEW

File: core/arch/arm/kernel/otp_stubs.c [hunk 1]
```diff
@@ -29,3 +29,20 @@ __weak int tee_otp_get_die_id(uint8_t *buffer, size_t len)
...
+ */
+__weak TEE_Result tee_otp_check_rpmb_key_write_lock(void)
+{
+	return TEE_SUCCESS;
+}
+
+__weak TEE_Result tee_otp_set_rpmb_key_write_lock(void)
+{
+	return TEE_SUCCESS;
+}
```
Comment (dmcilvaney) [inline core/arch/arm/kernel/otp_stubs.c]: +newline

File: core/drivers/fsl_sec/hw_key_blob.c [hunk 2]
```diff
@@ -235,3 +235,86 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
...
+	ccg_orig = sec_in32(ptov(CCM_BASE + MX6Q_CCM_CCGR2));
+	sec_out32(ptov(CCM_BASE + MX6Q_CCM_CCGR2), ccg_orig | OCOTP_CCG_ENABLE);
+
+#ifdef CFG_MX6
+	// 0x3E770000 is the OTP unlock key
+	// 0x26 is Bank 4 word 6, 0x27 is Bank 4 Word 7
+#if RPMB_KEY_WRITE_LOCK_FUSE_WORD == OCOTP_GP1_OFFSET
+	val = 0x3E770026;
+#elif RPMB_KEY_WRITE_LOCK_FUSE_WORD == OCOTP_GP2_OFFSET
+abcd
```
Comment (dmcilvaney) [inline core/drivers/fsl_sec/hw_key_blob.c]: Left over "is this actually compiling?" junk?

Comment (hbeberman) [reply core/drivers/fsl_sec/hw_key_blob.c]: Indeed it is.

**Comment by dmcilvaney** [comment]
If we are going to upstream this (I don't see a reason not to), we should update the readme etc relating to RPMB:
mk\config.mk: 135
documentation\secure_storage_rpmb.md:

**Pull request merged**